virttest/qemu_vm.py

```diff
@@ -998,6 +998,8 @@ def add_memorys(devices, params):
                 if params.get("hugepage_path"):
                     backend_options["backend"] = "memory-backend-file"
                     backend_options["mem-path"] = params["hugepage_path"]
+                    if params.get("share_mem") == "yes":
```

Comment:
Since we used `object_params` above, so here can be `share = yes`, but I am not familiar with this usage.
@huangyum , can you have a look?

Reply:
The param name should be 'share_mem_${name}' to specify the share option only for object 'name'. The 'name' is got from backend_id in https://github.com/avocado-framework/avocado-vt/blob/51794f7903a8876e36149c0c937ec95be9a85891/virttest/qemu_devices/qcontainer.py#L1443.
So in this case, the param in cfg should be `share_mem_machine_mem`.

Reply:
Since there comes a new requirement for the machine object now, I'm wondering maybe we could extend the code to allow more option configurations. Like, use object_params(name) to filter the options, then add them to `backend_options`.

Reply:
Thanks a lot, corrected.